lib/coverband/collectors/coverage.rb

```diff
@@ -82,27 +79,16 @@ def track_file?(file)
 
       private
 
-      def add_filtered_files(new_results)
-        new_results.each_pair do |file, line_counts|
-          next unless track_file?(file)
-          add_file(file, line_counts)
-        end
+      def filtered_files(new_results)
```

Comment:
@danmayer avoid the use of the instance variable here by just doing it all in one method that returns the filtered files.

Reply:
nice a little harder to mentally unpack but really concise and likely better memory / perf